### PR TITLE
chore(APIApplicationCommandInteractionDataOption): add types and their values

### PR DESCRIPTION
**Pull request by Benricheson101**
The current types don't throw an error if you include both `value` and `options`, this will.

### CODE REVIEW

**Comment by vladfrangu** [comment]
This is something I wanted done, so thanks! However, this would really benefit from https://github.com/discord/discord-api-docs/issues/2383, to strictly type it "correctly" (since it's always one property that we KNOW the value of)

**Comment by advaith1** [comment]
the type is now provided!
```js
options: [ { value: '443196999450361856', type: 6, name: 'user' } ]
```

**Comment by vladfrangu** [comment]
@Benricheson101 will you add the new types or would you want me to add them to your PR? 👀

**Comment by Benricheson101** [comment]
I can add them tomorrow

File: deno/v8/payloads/interactions.ts
Comment (advaith1) [inline deno/v8/payloads/interactions.ts:119]: 🤔 is this assuming that if there's no `type` then it's always a subcommand/group? currently, all options for commands sent on mobile don't have a `type` but in the future, all options will have a `type` afaik

File: deno/v8/payloads/interactions.ts
Comment (advaith1) [inline deno/v8/payloads/interactions.ts:133]: separate into one for `string`s and one for `Snowflake`s

**Review by advaith1 — CHANGES_REQUESTED** [review]

File: deno/v8/payloads/interactions.ts
Comment (Benricheson101) [reply deno/v8/payloads/interactions.ts:133]: I didn't realize there was as Snowflake type, I'll do that.

File: deno/v8/payloads/interactions.ts
Comment (Benricheson101) [reply deno/v8/payloads/interactions.ts:119]: That first line shouldn't be there. For the mobile thing, are you suggesting making `type` nullable for now?

Comment (advaith1) [reply deno/v8/payloads/interactions.ts:119]: or ig just remove that line and wait until it's fully released to merge this?

File: v8/payloads/interactions.ts
Comment (vladfrangu) [inline v8/payloads/interactions.ts:119]: Please split all these like the Command options ones :3

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/v8/payloads/interactions.ts
Comment (Benricheson101) [reply deno/v8/payloads/interactions.ts]: > That first line shouldn't be there.
Fixed this in 69ab92a

For the mobile `type` thing, someone said it would be rolled out this week, so it's probably not worth removing for just a few days.

File: v8/payloads/interactions.ts
Comment (vladfrangu) [reply v8/payloads/interactions.ts]: This wasn't resolved yet it was marked as such 👀

Comment (Benricheson101) [reply v8/payloads/interactions.ts]: That line [wasn't supposed to be there](https://github.com/discordjs/discord-api-types/pull/72#discussion_r575765745) and was removed in 9318e8d :thinking: What did you want me to do with it?

Comment (vladfrangu) [reply v8/payloads/interactions.ts]: > Please split all these like the Command options ones :3

I think this is clear enough... Take all possibilities for the DataOption and make them interfaces, just like the CommandOption ones

Comment (Benricheson101) [reply v8/payloads/interactions.ts]: Ohh sorry I misunderstood what you were asking, I'll do that.

**Comment by vladfrangu** [comment]
Hey @Benricheson101, what's the status? 😄 If you need someone to take over, let us know!

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**
